Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go

```diff
@@ -945,6 +989,22 @@ func (kl *Kubelet) registerWithApiserver() {
 				kl.registrationCompleted = true
 				return
 			}
+
+			if kl.cloud == nil {
+				// In the case where we're not using a cloud provider, the ExternalID could either have
+				// been the NodeIP, or the actual hostname.  If we switch between the two, consider it
```

Comment:
not in the code, fine to link from this PR, which is reachable from the commit breadcrumb